kore/test/Test/Kore/Internal/Pattern.hs

```diff
@@ -240,12 +240,9 @@ test_hasSimplifiedChildren =
     , testCase "One child isn't simplified, nested ands" $ do
         let simplified = Simplified_ Fully Any
             predicate =
-                makeAndPredicate
-                    (setSimplifiedPred simplified mockPredicate1)
-                    ( makeAndPredicate
+                    makeAndPredicate
```

Comment:
This assertion
```haskell
assertEqual
    "Children aren't simplified"
    False
    (Pattern.hasSimplifiedChildren topSideCondition patt)
```
is flaky, due to the `Eq` instance for `Predicates`, which equates predicates even when they have different Attributes:
https://github.com/runtimeverification/haskell-backend/blob/e936e5bd912b05347b7da18463ee17d535b16c49/kore/src/Kore/Internal/Predicate.hs#L285-L292

This means that the order we insert the list of predicates into the hash set decides if the test is true or false, since the list of predicates contains:

```
Predicate {... simplified = Simplified (SimplifiedData {sType = Fully ...} :< x,
Predicate {... simplified = Simplified (SimplifiedData {sType = Partly ...} :< x
```
and depending on which gets added first, this check from within `hasSimplifiedChildren` returns true or false:
https://github.com/runtimeverification/haskell-backend/blob/e936e5bd912b05347b7da18463ee17d535b16c49/kore/src/Kore/Internal/Pattern.hs#L137

because `isSimplified` matches on the `Fully/Partly` constructor for `true/false`

Reply:
@ana-pantilie ive modified the test to pass but do we want to address this somehow? I've tried changing the Eq instance to include the attributes but that made many more tests fail.

Reply:
Discussed during the meeting: we should remove this test because it wasn't testing what it's supposed to, and we don't even think it's testing a feature which would be useful.